code/modules/admin/admin.dm

```diff
@@ -2420,31 +2420,49 @@ var/global/noir = 0
 								tgui_alert(usr,"The game hasn't started yet!")
 								return
 
-							var/which_traitor = input("What kind of traitor?","Everyone's a Traitor") as null|anything in list(ROLE_TRAITOR,ROLE_WIZARD,ROLE_CHANGELING,ROLE_WEREWOLF,ROLE_VAMPIRE,ROLE_ARCFIEND,ROLE_HUNTER,ROLE_WRESTLER,ROLE_GRINCH,ROLE_OMNITRAITOR)
-							if(!which_traitor)
+							var/list/antag_options = list()
+							var/list/eligible_antagonist_types = concrete_typesof(/datum/antagonist) - (concrete_typesof(/datum/antagonist/subordinate) + concrete_typesof(/datum/antagonist/generic))
+							for (var/V as anything in eligible_antagonist_types)
+								var/datum/antagonist/A = V
+								antag_options[initial(A.display_name)] = initial(A.id)
+
+							var/selected_keyvalue = tgui_input_list(usr, "Choose an antagonist role to assign to everyone.", "Make Everyone An Antagonist", antag_options)
+							if (!selected_keyvalue)
 								return
-							var/hardmode = null
-							if (which_traitor == ROLE_TRAITOR)
-								if (tgui_alert(usr,"Hard Mode?","Everyone's a Traitor",list("Yes", "No")) == "Yes")
-									hardmode = "hardmode"
-							var/custom_objective = input("What should the objective be?","Everyone's a Traitor") as null|text
+
+							var/antagonist_role_id = antag_options[selected_keyvalue]
+
+							var/equip_traitor = TRUE
+							if (antagonist_role_id == ROLE_TRAITOR)
+								if (tgui_alert(usr, "Hard Mode?", "Make Everyone An Antagonist", list("Yes", "No")) == "Yes")
+									equip_traitor = FALSE
+
+							var/custom_objective = tgui_input_text(usr, "What should their objective be?", "Make Everyone An Antagonist")
 							if (!custom_objective)
 								return
-							var/escape_objective = input("Which escaping objective?") as null|anything in typesof(/datum/objective/escape/) + "None"
+							var/escape_objective = tgui_input_list(usr, "What should their escape objective be?", "Make Everyone An Antagonist", typesof(/datum/objective/escape/) + "None")
 							if (!escape_objective)
 								return
 
 							if (escape_objective == "None")
 								escape_objective = null
 
-							for(var/mob/living/carbon/human/H in mobs)
-								if(isdead(H) || !(H.client)) continue
-								if(checktraitor(H)) continue
-								evilize(H, which_traitor, hardmode, custom_objective, escape_objective)
+							for (var/mob/living/carbon/human/H in mobs)
+								if (isdead(H) || !H.mind || !H.client)
```

Comment:
```suggestion
								if (isdead(H) || !H.mind || !H.key)
```
I think this is more accurate, since we want to hit logged out players but not NPCs?